src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs

```diff
@@ -286,7 +286,7 @@ public void FilterMutants()
             }
 
             var skippedMutants = _input.ProjectInfo.ProjectContents.ReadOnlyMutants.Where(m => m.ResultStatus != MutantStatus.NotRun);
-            var skippedMutantGroups = skippedMutants.GroupBy(x => new { x.ResultStatus, x.ResultStatusReason }).OrderBy(x => x.Key.ResultStatusReason);
+            var skippedMutantGroups = skippedMutants.GroupBy(x => new { x.ResultStatus, x.ResultStatusReason }).OrderBy(x => x.Key.ResultStatus);
```

Comment:
This change means that skipped mutants that have the same result status reason (eg when using dashboard compare to reuse previous results) but different result statuses will no longer be ordered together.

@richardwerkman would you prefer

```
[21:34:40 INF] 11    mutants got status Ignored.      Reason: Removed by exclude from code coverage filter
[21:34:40 INF] 25    mutants got status Ignored.      Reason: Result based on previous run
[21:34:40 INF] 15    mutants got status NoCoverage.   Reason: Mutant has no coverage
[21:34:40 INF] 16    mutants got status NoCoverage.   Reason: Result based on previous run
[21:34:40 INF] 201   mutants got status Killed.       Reason: Result based on previous run
[21:34:40 INF] 100   mutants got status Survived.     Reason: Result based on previous run
[21:34:40 INF] 10    mutants got status CompileError. Reason: Result based on previous run
[21:34:40 INF] 3     mutants got status Timeout.      Reason: Result based on previous run
```

Or:

```
[21:34:40 INF] 11    mutants got status Ignored.      Reason: Removed by exclude from code coverage filter
[21:34:40 INF] 15    mutants got status NoCoverage.   Reason: Mutant has no coverage
[21:34:40 INF] 201   mutants got status Killed.       Reason: Result based on previous run
[21:34:40 INF] 16    mutants got status NoCoverage.   Reason: Result based on previous run
[21:34:40 INF] 25    mutants got status Ignored.      Reason: Result based on previous run
[21:34:40 INF] 100   mutants got status Survived.     Reason: Result based on previous run
[21:34:40 INF] 10    mutants got status CompileError. Reason: Result based on previous run
[21:34:40 INF] 3     mutants got status Timeout.      Reason: Result based on previous run
```

Reply:
I think it's both a bit much. We could also go for something like:

```
[21:34:40 INF] Based on your previous run 336 mutants go a status without testing (201 Killed, 100 Survived, 11 Ignored, 16 NoCoverage, 10 CompileError, 3 Timeout)
```
Also, is the CompileError really based on the previous run? I think we get the errors again before we use the baseline so we could skip them in printing here?

Reply:
That means specifically looking for the string 'Result based on previous run' and changing the output based on it. Currently this whole reporting piece is generic for all filters.

It only shows up if you use dashboard compare, and I would think a user would want clear information about what exactly was picked up by the comparison, especially since we've seen some small errors with this feature.

Reply:
I see, the grouping appeared a bit strange to me. I can revert, but the problem is that the algorithm now reports non injected mutants as CompileError, so the sorting by 'Reason' first resulted in having them separated from actual compileError.

I am not sure what the proper design is, but it sounds to me that the 'reason' field has many different usage then

Reply:
Can you give an example of the output and expected output?

Reply:
As hinted earlier, I am not sure what the output should look like. For context: I have never tested the diff mode, so I never encountered the 'Result based on previous run' reason. Also, I was under the impression that 'StatusReason' purpose was to describe how Stryker reached to that status.
I think this deserve more discussion before reaching a conclusion, so I restored the order as it was, and I will open an issue for discussion.

Here is a simulated output of what can reported now, and I think this looks messy as 'CompileError' status is all over the place.

```[10:55:29 INF] Capture mutant coverage using 'perTest' mode.
[10:55:41 INF] 32    mutants got status CompileError. Reason: Mutant caused compile errors
[10:55:41 INF] 2     mutants got status NoCoverage.   Reason: Mutant has no test coverage
[10:55:41 INF] 11    mutants got status Ignored.      Reason: Removed by exclude from code coverage filter
[10:55:41 INF] 10    mutants got status CompileError. Reason: Result based on previous run
[10:55:41 INF] 201   mutants got status Killed.       Reason: Result based on previous run
[10:55:41 INF] 16    mutants got status NoCoverage.   Reason: Result based on previous run
[10:55:41 INF] 25    mutants got status Ignored.      Reason: Result based on previous run
[10:55:41 INF] 100   mutants got status Survived.     Reason: Result based on previous run
[10:55:41 INF] 3     mutants got status CompileError. Reason: Stryker was not able to inject mutation in code.
[10:55:41 INF] xxx   total mutants are skipped for the above mentioned reasons
[10:55:41 INF] yyyy  total mutants will be tested```

Reply:
I agree, we should think about what we're gonna do with this 👍